core/jvm/src/main/scala/zio/internal/metrics/ConcurrentMetricHooksPlatformSpecific.scala

```diff
@@ -175,4 +190,19 @@ class ConcurrentMetricHooksPlatformSpecific extends ConcurrentMetricHooks {
 
     MetricHook(update, () => MetricState.Frequency(snapshot()))
   }
+
+  private final class AtomicMinMax {
+    private val minMax = new AtomicReference(Option.empty[(Double, Double)])
+
+    def get(): (Double, Double) = minMax.get().getOrElse(0.0 -> 0.0)
+
+    def update(value: Double): Unit =
```

Comment:
This will involve multiple allocations per `update`. We are trying to achieve "zero allocations" on the happy path.

In order to do this, you can factor `minMax` instead as 2 atomic doubles: one to hold the min, and one to hold the max. Then, I would implement the atomic logic with a `@tailrec` or with a `while` loop, so as to avoid allocation of the function passed to `updateAndGet`.

Reply:
You can use a sentinel value to denote absence of being set: for min, it can be `Double.MaxValue`; for max, it can be `Double.MinValue`.

Reply:
That's how I had it initially :) Then I went with the current approach.

Thanks for suggesting the Double sentinels, I was initially using `0.0` but then realized we could have negative values.  I think that is what led me to the current solution.